### PR TITLE
feat: node-cache instrumentation

### PR DESCRIPTION
**Pull request by nirsky**

### CODE REVIEW

File: packages/instrumentation-node-cache/src/node-cache.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import type * as NodeCache from 'node-cache';
```
Comment (nirsky) [inline packages/instrumentation-node-cache/src/node-cache.ts:1]: @blumamir I know the recommendation is to have to types as a dependency (and not devDep).  
But in this case, the types are part of the main lib, any recommendation here?

File: packages/instrumentation-node-cache/package.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        "@opentelemetry/instrumentation": "^0.22.0",
+        "@opentelemetry/semantic-conventions": "^0.22.0"
+    },
+    "devDependencies": {
+        "@opentelemetry/api": "^1.0.0",
+        "@opentelemetry/tracing": "^0.22.0",
+        "@types/mocha": "^8.2.2",
+        "expect": "^26.6.2",
+        "mocha": "^8.4.0",
+        "node-cache": "^5.1.2",
```
Comment (habmic) [inline packages/instrumentation-node-cache/package.json:55]: Since we support 5.0.0 and above, shouldn't it be also 5.0.0 ?

**Review by habmic — APPROVED** [review]

File: packages/instrumentation-node-cache/package.json (hunk 2 above)
Comment (nirsky) [reply packages/instrumentation-node-cache/package.json:55]: I think it's ok to pick one specific version for development purposes.
In the ci we have test-all-versions which checks all the `>=5.0.0` versions

File: packages/instrumentation-node-cache/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export interface NodeCacheRequestInfo {
+    moduleVersion: string;
+    operation: string;
+    args: any[];
+}
+
+export interface NodeCacheInstrumentationConfig extends InstrumentationConfig {
+    requestHook?: (span: Span, requestInfo: NodeCacheRequestInfo) => void;
+    responseHook?: (span: Span, response: any) => void;
```
Comment (blumamir) [inline packages/instrumentation-node-cache/src/types.ts:12]: I think that if we use `requestInfo` we should also use `responseInfo`, even if it currently only contains the response object

File: packages/instrumentation-node-cache/src/node-cache.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                    kind: SpanKind.INTERNAL,
+                    attributes: {
+                        [SemanticAttributes.DB_SYSTEM]: 'node-cache',
+                        [SemanticAttributes.DB_OPERATION]: opName,
+                        [SemanticAttributes.DB_STATEMENT]: toStatement(arguments),
+                    },
+                });
+
+                if (self._config.requestHook) {
+                    try {
```
Comment (blumamir) [inline packages/instrumentation-node-cache/src/node-cache.ts:94]: like on not using `safeExecuteInTheMiddle`

File: packages/instrumentation-node-cache/src/node-cache.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                        self._config.requestHook(span, {
+                            moduleVersion,
+                            operation: opName,
+                            args: Object.values(arguments),
```
Comment (blumamir) [inline packages/instrumentation-node-cache/src/node-cache.ts:98]: isn't arguments an array? why do we need to call `Object.values`?

File: packages/instrumentation-node-cache/src/node-cache.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                        } catch (err) {
+                            diag.error('node-cache instrumentation: responseHook error', err);
+                        }
+                    }
+                    return response;
+                } catch (err) {
+                    span.recordException(err);
+                    span.setStatus({
+                        code: SpanStatusCode.ERROR,
+                        message: err.message,
```
Comment (blumamir) [inline packages/instrumentation-node-cache/src/node-cache.ts:123]: `err?.message`. can `err` be of type `string` here?

**Review by blumamir — APPROVED** [review]

File: packages/instrumentation-node-cache/src/node-cache.ts (hunk 5 above)
Comment (nirsky) [reply packages/instrumentation-node-cache/src/node-cache.ts:98]: it's an array-like object (implementing IArguments interface).   
You can use it like array but typescript didn't approve this

File: packages/instrumentation-node-cache/src/node-cache.ts (hunk 6 above)
Comment (nirsky) [reply packages/instrumentation-node-cache/src/node-cache.ts:123]: fixed

File: packages/instrumentation-node-cache/src/types.ts (hunk 3 above)
Comment (nirsky) [reply packages/instrumentation-node-cache/src/types.ts]: fixed

**Pull request merged**